src/fides/api/ops/api/v1/endpoints/messaging_endpoints.py

```diff
@@ -135,6 +146,63 @@ def patch_config_by_key(
         )
 
 
+@router.put(
+    MESSAGING_DEFAULT,
+    dependencies=[Security(verify_oauth_client, scopes=[MESSAGING_CREATE_OR_UPDATE])],
+    response_model=MessagingConfigResponse,
+)
+def put_default_config(
+    *,
+    db: Session = Depends(deps.get_db),
+    messaging_config: MessagingConfigBase,
+) -> Optional[MessagingConfigResponse]:
+    """
+    Updates default messaging config for given service type.
+    """
+    logger.info(
+        "Starting upsert for default messaging config of type '{}'",
+        messaging_config.service_type,
+    )
+    incoming_data = messaging_config.dict()
+    existing_default = MessagingConfig.get_by_type(db, messaging_config.service_type)
+    if existing_default:
+        # take the key of the existing default and add that to the incoming data, to ensure we overwrite the same record
+        incoming_data["key"] = existing_default.key
```

Comment:
even if we have an existing default, we generate the same key / name if it's the same service, right? So this code is just to be extra safe?

Reply:
yeah, generally that should be the case. but to keep backward compatibility, the old messaging endpoints (that allow for name/key specification) are still available, and any messaging configs created by these means will also be exposed as a "default". effectively, all messaging configs are defaults now :)